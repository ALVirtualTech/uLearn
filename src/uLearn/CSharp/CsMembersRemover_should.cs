using System;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using uLearn.Model.Blocks;

namespace uLearn.CSharp
{
	[TestFixture]
	public class CsMembersRemover_should
	{
		private readonly FileSystem fs = new FileSystem(new DirectoryInfo("tests"));

		private string LoadCode(string file = "OverloadedMethods.cs")
		{
			return fs.GetContent(file);
		}

		private string RemoveRegion(string region, bool onlyBody = false)
		{
			var code = LoadCode();
			new CsMembersRemover().Remove(ref code, new[] { new Label { Name = region, OnlyBody = onlyBody } });
			return code;
		}

		private string RemoveRegions(params Label[] regions)
		{
			var code = LoadCode();
			new CsMembersRemover().Remove(ref code, regions);
			return code;
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleMethod()
		{
			Approvals.Verify(RemoveRegion("Main"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleMethodBody()
		{
			Approvals.Verify(RemoveRegion("Main", true));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void OverloadMethod()
		{
			Approvals.Verify(RemoveRegion("Method"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void OverloadMethodBody()
		{
			Approvals.Verify(RemoveRegion("Method", true));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleClass()
		{
			Approvals.Verify(RemoveRegion("OverloadedMethods"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleClassBody()
		{
			Approvals.Verify(RemoveRegion("OverloadedMethods", true));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleMethodSolution()
		{
			var code = LoadCode();
			var index = new CsMembersRemover().RemoveSolution(ref code, new Label { Name = "Main" });
			Approvals.Verify(String.Format("solution index: {0}\r\nCode:\r\n{1}", index, code));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void SingleMethodBodySolution()
		{
			var code = LoadCode();
			var index = new CsMembersRemover().RemoveSolution(ref code, new Label { Name = "Main", OnlyBody = true });
			Approvals.Verify(String.Format("solution index: {0}\r\nCode:\r\n{1}", index, code));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void RemoveUsings()
		{
			var code = LoadCode();
			Approvals.Verify(CsMembersRemover.RemoveUsings(code));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void RemoveManyMethods()
		{
			Approvals.Verify(RemoveRegions(new Label { Name = "Method" }, new Label { Name = "Main" }));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void RemoveManyMethodsSameBody()
		{
			Approvals.Verify(RemoveRegions(new Label { Name = "Method", OnlyBody = true }, new Label { Name = "Main" }));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void RemoveManyMethodsAllBody()
		{
			Approvals.Verify(RemoveRegions(new Label { Name = "Method", OnlyBody = true }, new Label { Name = "Main", OnlyBody = true }));
		}
	}
}