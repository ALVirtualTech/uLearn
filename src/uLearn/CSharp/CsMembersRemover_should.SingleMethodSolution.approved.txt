solution index: 415
Code:
//region all

using System;
using System.IO;

namespace uLearn.tests
{
	internal class OverloadedMethods : IRunnable
	{
		//region methods_1_and_2
		public string Method()
		{
			return "a";
		}

		//region methods_2_and_3

		public string Method(int a)
		{
			return a.ToString("0");
		}

		//endregion methods_1_and_2

		private string Method(string a)
		{
			return a;
		}

#line 1
	}
}

//endregion all